w3af/plugins/auth/detailed.py

```diff
@@ -58,7 +60,24 @@ def login(self):
 
         try:
             functor = getattr(self._uri_opener, self.method)
-            functor(self.auth_url, data)
+
+            redir_count = 0
+
+            while True:
+                if redir_count > 5:
+                    break
+                # actually file the auth HTTP reuqest
+                r = functor(self.auth_url, data)
+
+                # follow redirects if the feature is enabled
+                if self.follow_redirects and r.get_code() in GET_HEAD_CODES:
+                    self.auth_url.set_path(r.get_headers().iget('location')[0])
+                    om.out.debug('Auth redirected to new URL', self.auth_url)
+                    self.method = 'GET'
+                    data = ''
+                    redir_count += 1
+                else:
+                    break
```

Comment:
This is a big improvement from the past PR, thanks for fixing some of the issues that I pointed out (such as the loop). These issues are still pending:
 * You're modifying the user-configured setting `self.auth_url` and `self.method` which is not desirable
 * You're only supporting "location" header instead of all the ones which the framework knows about (used in `web_spider.py:_headers_url_generator`). Maybe you could re-factor the code used in the web_spider.py into a `core` function that both the web_spider and detailed plugin could use? Something like "extract_redirects(http_response)" which returns a list of all the URL instances which the browser would be redirected to. I know that in 99% of the cases this would be a list with only one URL, but we never know...